webview/src/plots/components/webviewSlice.ts

```diff
@@ -38,17 +36,12 @@ export const webviewSlice = createSlice({
     initialize: (state: { hasData: boolean }) => {
       state.hasData = true
     },
-    setSnapPoints: (
-      state: { snapPoints: SnapPoints },
+    setMaxNbPlotsPerRow: (
+      state: { maxNbPlotsPerRow: number },
       action: PayloadAction<number>
     ) => {
-      // Action payload here is the max width of the plots webview. When changed, we re-calculate the snap points.
-      state.snapPoints = [
-        action.payload,
-        action.payload / 2,
-        action.payload / 3,
-        action.payload / 4
-      ]
+      // Action payload here is the max width of the plots webview. When changed, we re-calculate the max number of plots per row
+      state.maxNbPlotsPerRow = Math.floor(action.payload / 300)
```

Comment:
[N] [for completeness](https://github.com/iterative/vscode-dvc/pull/3405/files#diff-66b700260ad828bfc40b1f51e751b4cf4798759be95acb9555e81a77588d59c4R39) from this could be

```
const width = action.payload
state.maxNbPlotsPerRow = Math.floor((width - 100) / 300)
```